commands/snippet/create/create.go

```diff
@@ -0,0 +1,120 @@
+package create
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/profclems/glab/api"
+	"github.com/profclems/glab/commands/cmdutils"
+	"github.com/profclems/glab/internal/glrepo"
+	"github.com/profclems/glab/pkg/iostreams"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+)
+
+type CreateOpts struct {
+	Title       string
+	Description string
+	Filename    string
+	Visibility  string
+
+	ForUser bool
+
+	IO       *iostreams.IOStreams
+	Lab      func() (*gitlab.Client, error)
+	BaseRepo func() (glrepo.Interface, error)
+}
+
+func NewCmdCreate(f *cmdutils.Factory) *cobra.Command {
+	opts := &CreateOpts{}
+	var snippetCreateCmd = &cobra.Command{
+		Use:     "create",
```

Comment:
We should allow an optional file path to be specified as an argument `create [path]`. When no path is specified as argument, we fall back to stdin. `--filename` should be optional when `path` is specified so we should the filename as default.

This will help us when the API finally supports multiple snippet files.

Reply:
You right. Gonna try to get these changes in